### PR TITLE
Refactor profiles in remaining queries referencing Profiles

### PR DESCRIPTION
**Pull request by Rotorsoft**
## Link to Issue
Closes: #8472

## Description of Changes
- Replaces Profiles with Users.profile JSONB in queries
- A little bit of code cleanup and optimization
- Creates query and comment schemas, shared by client and server

 
## Test Plan
- Pass tests
- @masvelio & @mzparacha to review UI changes

### CODE REVIEW

**Review by timolegros — COMMENTED** [review]
Generally works/looks good on first pass but need to resolve conflicts and merge master before I can test more deeply the feeds.

**Comment by mzparacha** [comment]
I tried some profile/thread related user flows, found some issues

1. When creating a thread, the username is always `Anonymous`, even when profile has username

https://github.com/user-attachments/assets/72d271c7-08e6-443d-92a4-b85117b9699e

2. New comments also have `Anonymous` names
image


2. Not sure if its planned for this PR, but the edit thread route still returns User.Profiles
image

**Comment by mzparacha** [comment]
Per the changes done, I tested these user flows

1. Thread CRUD
2. Comments CRUD
3. Thread/Comment Reactions CRUD
4. User Profile CRUD
5. User Addresses CRUD (address linking etc)
6. Account creation flows (as we updated some passort.js logic)

and found this issue

1. Can't use the edit thread API to edit/mark or unmark spam/update status/add or remove collaborators and other actions. The API fails with this error
```
{"status":400,"error":"Must provide body"}
```

the other things I tested worked for me.

cc: @Rotorsoft

**Comment by timolegros** [comment]
> Per the changes done, I tested these user flows
> 
> 1. Thread CRUD
> 2. Comments CRUD
> 3. Thread/Comment Reactions CRUD
> 4. User Profile CRUD
> 5. User Addresses CRUD (address linking etc)
> 6. Account creation flows (as we updated some passort.js logic)
> 
> and found this issue
> 
> 1. Can't use the edit thread API to edit/mark or unmark spam/update status/add or remove collaborators and other actions. The API fails with this error
> 
> ```
> {"status":400,"error":"Must provide body"}
> ```
> 
> the other things I tested worked for me.
> 
> cc: @Rotorsoft

@kurtisassad might be caused by #8041? Specifically https://github.com/hicommonwealth/commonwealth/blob/master/packages/commonwealth/server/controllers/server_threads_methods/update_thread.ts#L110.

**Comment by Rotorsoft** [comment]
> > 1. Can't use the edit thread API to edit/mark or unmark spam/update status/add or remove collaborators and other actions. The API fails with this error
> > 
> > ```
> > {"status":400,"error":"Must provide body"}
> > ```
> 
> @kurtisassad might be caused by #8041? Specifically https://github.com/hicommonwealth/commonwealth/blob/master/packages/commonwealth/server/controllers/server_threads_methods/update_thread.ts#L110.

@kurtisassad can you check this? ...seems related to the history PR

File: packages/commonwealth/client/scripts/models/Thread.ts [hunk 1]
```diff
@@ -464,6 +464,7 @@ export class Thread implements IUniqueId {
       associatedContests,
       ContestActions,
     );
+    console.log(recentComments);
```
Comment (kurtisassad) [inline packages/commonwealth/client/scripts/models/Thread.ts:467]: We should remove the log

File: packages/commonwealth/test/integration/api/getAddressProfile.spec.ts [hunk 2]
```diff
@@ -95,20 +83,21 @@ describe('getAddressProfile tests', () => {
...
       ],
-    } as GetAddressProfileReq;
+    };
 
     const resp = await getAddressProfiles(models, postReq(r), res());
 
     chai.assert.equal(resp['result'].length, 3);
 
     const results = resp['result'];
+    console.log(results);
```
Comment (kurtisassad) [inline packages/commonwealth/test/integration/api/getAddressProfile.spec.ts:100]: here too

**Review by kurtisassad — APPROVED** [review]

**Review by rbennettcw — APPROVED** [review]

**Pull request merged**